opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -349,6 +362,334 @@ def collect(
         return current_point
 
 
+# pylint: disable=protected-access
+class _ExponentialBucketHistogramAggregation(_Aggregation[HistogramPoint]):
+    # _min_max_size and _max_max_size are the smallest and largest values
+    # the max_size parameter may have, respectively.
+
+    # _min_max_size is is the smallest reasonable value which is small enough
+    # to contain the entire normal floating point range at the minimum scale.
+    _min_max_size = 2
+
+    # _max_max_size is an arbitrary limit meant to limit accidental creation of
+    # giant exponential bucket histograms.
+    _max_max_size = 16384
+
+    def __init__(
+        self,
+        attributes: Attributes,
+        start_time_unix_nano: int,
+        # This is the default maximum number of buckets per positive or
+        # negative number range.  The value 160 is specified by OpenTelemetry.
+        # See the derivation here:
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exponential-bucket-histogram-aggregation)
+        max_size: int = 160,
+    ):
+        super().__init__(attributes)
+        # max_size is the maximum capacity of the positive and negative
+        # buckets.
+        if max_size < self._min_max_size:
+            raise Exception("size {max_size} is smaller than {min_max_size}")
+
+        if max_size > self._max_max_size:
+            raise Exception("size {max_size} is larter than {max_max_size}")
```

Comment:
These error messages should be more descriptive about what invariant is being violated, "size 0 is less than 2" is not very descriptive.

I would also recommend throwing [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)

Reply:
Right, fixed.